File: Sources/ComprovacaoFacilLattes/Models/LattesProfile.swift
import Foundation
import SwiftData

@Model
final class LattesProfile {
    @Attribute(.unique) var id: UUID
    var name: String
    var pdfPath: String
    var importDate: Date
    var lastUpdated: Date
    var savePath: String
    var rawText: String

    /// Aprendizado: vínculos recusados pelo usuário ("nomeArquivo||hashEntrada").
    /// Em novos escaneamentos, essas combinações são evitadas.
    var rejectedLinks: [String] = []

    @Relationship(deleteRule: .cascade, inverse: \LattesSection.profile)
    var sections: [LattesSection]

    // Todos os certificados do perfil (assigned + limbo)
    @Relationship(deleteRule: .cascade, inverse: \Certificate.profile)
    var certificates: [Certificate]

    init(name: String, pdfPath: String, savePath: String) {
        self.id = UUID()
        self.name = name
        self.pdfPath = pdfPath
        self.savePath = savePath
        self.rawText = ""
        self.importDate = Date()
        self.lastUpdated = Date()
        self.sections = []
        self.certificates = []
    }

    var sortedSections: [LattesSection] {
        sections.sorted { $0.order < $1.order }
    }

    var totalEntries: Int {
        sections.reduce(0) { $0 + $1.entries.count }
    }

    var confirmedCount: Int {
        sections.reduce(0) { acc, s in
            acc + s.entries.filter { $0.certificateStatus == .confirmed }.count
        }
    }

    var suggestedCount: Int {
        sections.reduce(0) { acc, s in
            acc + s.entries.filter { $0.certificateStatus == .suggested }.count
        }
    }

    var pendingCount: Int {
        sections.reduce(0) { acc, s in
            acc + s.entries.filter { $0.certificateStatus == .none }.count
        }
    }

    var limboCertificates: [Certificate] {
        certificates.filter { $0.entry == nil && !$0.isRejected }
    }
}

File: Versão Windows/src/ComprovacaoFacilLattes.Core/Models/LattesProfile.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ComprovacaoFacilLattes.Core.Models;

/// <summary>Um currículo/perfil — o app suporta vários em paralelo.</summary>
public class LattesProfile
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = "";
    public string PdfPath { get; set; } = "";
    public DateTime ImportDate { get; set; } = DateTime.UtcNow;
    public DateTime LastUpdated { get; set; } = DateTime.UtcNow;

    /// <summary>Pasta padrão sugerida para salvar o relatório e para onde comprovantes importados são copiados.</summary>
    public string SavePath { get; set; } = "";

    /// <summary>Texto completo extraído do PDF (cache, evita reabrir o PDF para exibir resumo).</summary>
    public string RawText { get; set; } = "";

    /// <summary>
    /// Aprendizado: <c>"nomeArquivoSemExtensao||hashKeyDaEntrada"</c> — vínculos que o usuário já
    /// recusou explicitamente, para não sugerir de novo.
    /// </summary>
    public List<string> RejectedLinks { get; set; } = new();

    public List<LattesSection> Sections { get; set; } = new();

    /// <summary>TODOS os certificados do perfil (vinculados + em limbo).</summary>
    public List<Certificate> Certificates { get; set; } = new();

    public LattesProfile() { }

    public LattesProfile(string name, string pdfPath, string savePath)
    {
        Name = name;
        PdfPath = pdfPath;
        SavePath = savePath;
    }

    [NotMapped]
    public IEnumerable<LattesSection> SortedSections => Sections.OrderBy(s => s.Order);

    [NotMapped]
    public int TotalEntries => Sections.Sum(s => s.Entries.Count);

    [NotMapped]
    public int ConfirmedCount => Sections.Sum(s => s.Entries.Count(e => e.CertificateStatus == EntryStatus.Confirmed));

    [NotMapped]
    public int SuggestedCount => Sections.Sum(s => s.Entries.Count(e => e.CertificateStatus == EntryStatus.Suggested));

    [NotMapped]
    public int PendingCount => Sections.Sum(s => s.Entries.Count(e => e.CertificateStatus == EntryStatus.None));

    /// <summary>Certificados escaneados sem vínculo encontrado, ou vínculo desfeito.</summary>
    [NotMapped]
    public IEnumerable<Certificate> LimboCertificates => Certificates.Where(c => c.Entry == null && !c.IsRejected);
}
